gqcp/include/QCModel/HF/StabilityMatrices/GHFStabilityMatrices.hpp

```diff
@@ -0,0 +1,258 @@
+
+        // Determine the dimensions of the total stability matrix.
+        const auto K = A.dimension(0);
+        const auto dim = 2 * K;
+
+        // Create the total stability matrix as specified above in the documentation.
+        Matrix H {dim, dim};
+
+        H.topLeftCorner(K, K) = A;
+        H.topRightCorner(K, K) = B;
+        H.bottomLeftCorner(K, K) = B.conjugate();
+        H.bottomRightCorner(K, K) = A.conjugate();
+
+        return H;
+    }
+
+
+    /*
+     *  MARK: Checking the stability
+     */
+
+    /**
+     *  @return a boolean, telling us whether or not the real valued internal stability matrix belongs to a stable or unstable set of parameters.
+     */
+    template <typename S = Scalar>
+    enable_if_t<std::is_same<S, double>::value, bool> isInternallyStable() const {
+
+        // The first step is to calculate the correct stability matrix: This method checks the internal stability of a real valued wavefunction.
+        const auto stability_matrix = this->Internal();
+
+        // Create an eigensolver to diagonalize the stability matrix.
+        using MatrixType = Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>;
+        Eigen::SelfAdjointEigenSolver<MatrixType> eigensolver {stability_matrix};
+
+        // Calculate the eigenvalues and check whether they are strictly positive or not, we update the stability properties accordingly.
+        const auto& eigenvalues = eigensolver.eigenvalues().transpose();
+
+        if (eigenvalues[0] < -1.0e-5) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+
+    /**
+     *  @return a boolean, telling us whether or not the real valued external stability matrix belongs to a stable or unstable set of parameters.
+     */
+    template <typename S = Scalar>
+    enable_if_t<std::is_same<S, double>::value, bool> isExternallyStable() const {
+
+        // The first step is to calculate the correct stability matrix: This method checks the internal stability of a real valued wavefunction.
+        const auto stability_matrix = this->External();
+
+        // Create an eigensolver to diagonalize the stability matrix.
+        using MatrixType = Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>;
+        Eigen::SelfAdjointEigenSolver<MatrixType> eigensolver {stability_matrix};
+
+        // Calculate the eigenvalues and check whether they are strictly positive or not, we update the stability properties accordingly.
+        const auto& eigenvalues = eigensolver.eigenvalues().transpose();
+
+        if (eigenvalues[0] < -1.0e-5) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+
+    /**
+     *  @return a boolean, telling us whether or not the complex valued internal stability matrix belongs to a stable or unstable set of parameters.
+     */
+    template <typename S = Scalar>
+    enable_if_t<std::is_same<S, complex>::value, bool> isInternallyStable() const {
```

Comment:
I think that, since the `this->internal()` API has `enable_if`, you can actually remove the `enable_if` on both the `isInternallyStable` methods. I think the compiler can redirect its `interal()` calls.

Reply:
If I remove the enable_ifon both isInternallyStable() methods, doesn't that mean I can remove one of them completely, as they will become identical?